verification/api/handler.go

```diff
@@ -0,0 +1,385 @@
+// Copyright 2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package api
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/denisbrodbeck/machineid"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/veraison/services/verification/sessionmanager"
+	"github.com/veraison/services/verification/verifier"
+)
+
+const (
+	ChallengeResponseSessionMediaType = "application/vnd.veraison.challenge-response-session+json"
+)
+
+var (
+	ErrInternal = errors.New("internal error")
+)
+
+var (
+	tenantID = "0123456789"
+)
+
+type IHandler interface {
+	NewChallengeResponse(c *gin.Context)
+	SubmitEvidence(c *gin.Context)
+	GetSession(c *gin.Context)
+}
+
+type Handler struct {
+	SessionManager sessionmanager.ISessionManager
+	Verifier       verifier.IVerifier
+}
+
+func NewHandler(sm sessionmanager.ISessionManager, v verifier.IVerifier) IHandler {
+	return &Handler{
+		SessionManager: sm,
+		Verifier:       v,
+	}
+}
+
+var (
+	ConfigNonceSize     uint8 = 32
+	ConfigSessionTTL, _       = time.ParseDuration("2m30s")
+)
+
+// mintSessionID creates a version 1 UUID based on a unique machine ID, clock
+// sequence and current time.  Routing to the correct node can therefore happen
+// based on the NodeID part of the UUID (i.e., octets 10-15).
+func mintSessionID() (uuid.UUID, error) {
+	mid, err := machineid.ID()
+	if err != nil {
+		return uuid.UUID{}, err
+	}
+
+	uuid.SetNodeID([]byte(mid))
+
+	return uuid.NewUUID()
+}
+
+// mintNonce creates a random nonce of nonceSz bytes.  nonceSz must be strictly
+// positive
+func mintNonce(nonceSz uint8) ([]byte, error) {
+	if nonceSz == 0 {
```

Comment:
We should check here for acceptable nonceSz right?

Reply:
I don't understand this comment, sorry.

Reply:
I mean Veraison does not accept any NonceSz but only few right, 32, 64 and 128

Reply:
oh, where did we write that?

Reply:
that'd be utterly wrong

Reply:
description:
            base64 encoded 32, 48, or 64 bytes random value